src/tools/networkTools.cpp

```diff
@@ -211,39 +210,36 @@ std::map<std::string, std::string> getNetworkInterfaces() {
   return result;
 }
 
-
-std::string getBestPublicIp(bool ipv6) {
-  IpAddress bestPublicIp = IpAddress{"127.0.0.1","::1"};
+IpAddress getBestPublicIps(IpMode mode) {
+  IpAddress bestPublicIps;
   std::map<std::string, IpAddress> interfaces = getNetworkInterfacesIPv4Or6();
-
 #ifndef _WIN32
   const char* const prioritizedNames[] = { "eth0", "eth1", "wlan0", "wlan1", "en0", "en1" };
-  for(auto name: prioritizedNames) {
-    auto it=interfaces.find(name);
-    if(it != interfaces.end() && !(ipv6 && (*it).second.addr6.empty())) {
-      bestPublicIp = (*it).second;
-      break;
-    }
+  for (const auto& name : prioritizedNames) {
+    const auto it = interfaces.find(name);
+    if (it == interfaces.end()) continue;
+    const IpAddress& interfaceIps = it->second;
+    if (bestPublicIps.addr.empty()) bestPublicIps.addr = interfaceIps.addr;
+    if (bestPublicIps.addr6.empty()) bestPublicIps.addr6 = interfaceIps.addr6;
   }
 #endif
-
-  const char* const prefixes[] = { "192.168", "172.16.", "10.0" };
-  for(auto prefix : prefixes){
-    for(auto& itr : interfaces) {
-      std::string interfaceIp(itr.second.addr);
-      if (interfaceIp.find(prefix) == 0 && !(ipv6 && itr.second.addr6.empty())) {
-        bestPublicIp = itr.second;
-        break;
-      }
+  const char* const v4prefixes[] = { "192.168", "172.16", "10.0", "169.254" };
+  for (const auto& prefix : v4prefixes) {
+    for (const auto& [_, interfaceIps] : interfaces) {
+      if (bestPublicIps.addr.empty() && interfaceIps.addr.find(prefix) == 0) bestPublicIps.addr = interfaceIps.addr;
+      if (bestPublicIps.addr6.empty()) bestPublicIps.addr6 = interfaceIps.addr6;
     }
   }
```

Comment:
I wonder if we should try matching via network address prefix before using looking at the network interface names. @kelson42 What's your opinion?